src/waveform/renderers/waveformrenderplaymarker.cpp

```diff
@@ -38,4 +56,36 @@ void WaveformRenderPlayMarker::draw(QPainter* painter, QPaintEvent* event) {
         painter->drawLine(0, lineY + 1, width, lineY + 1);
         painter->drawLine(0, lineY - 1, width, lineY - 1);
     }
+
+    // Draw bar and beat counter
+    if (prevBeat != mixxx::kInvalidBeat) {
+        QString barBeatString = QString("%1.%2").arg(
+                QString::number(prevBeat.getBarIndex() + 1),
+                QString::number(prevBeat.getBarRelativeBeatIndex() + 1));
+        painter->setPen(colors->getPlayPosColor());
+        painter->setBrush(colors->getBgColor());
+        painter->setOpacity(kBarBeatTextBoxOpacity);
+        const int boxHeightPixels = painter->fontMetrics().height();
+        const int boxWidthPixels = painter->fontMetrics().width(
+                barBeatString, barBeatString.length());
+        const int textPaddingPixels = 2;
+        if (orientation == Qt::Horizontal) {
+            painter->drawRect(lineX,
+                    height / 2 - boxHeightPixels / 2 - textPaddingPixels,
```

Comment:
Drawing this in the middle of the bertical height obscures the view of the waveform. I think it would be better at the top.